drivers/cpl/nuopc/glc_comp_nuopc.F90

```diff
@@ -0,0 +1,934 @@
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+    ! read in the mesh
+    EMesh = ESMF_MeshCreate(filename=trim(cvalue), fileformat=ESMF_FILEFORMAT_ESMFMESH, &
+          elementDistgrid=Distgrid,  rc=rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+    ! elemAreaArray = ESMF_ArrayCreate(DistGrid, mesh_areas, rc=rc)
+    ! if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    ! call ESMF_MeshGet(Emesh, elemAreaArray=elemAreaArray, rc=rc)
+    ! if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+    !--------------------------------
+    ! Realize the actively coupled fields
+    !--------------------------------
+
+    call realize_fields(importState, exportState, Emesh, flds_scalar_name, flds_scalar_num, rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+    !--------------------------------
+    ! Check consistency of mesh with internal CISM lats, lons, areas
+    !--------------------------------
+
+    ! obtain mesh lats, lons and areas
+    call ESMF_MeshGet(Emesh, spatialDim=spatialDim, numOwnedElements=numOwnedElements, rc=rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+    if (numOwnedElements /= npts) then
+       call shr_sys_abort('ERROR: numOwnedElements is not equal to npts')
+    end if
+
+    allocate(ownedElemCoords(spatialDim*numOwnedElements))
+    call ESMF_MeshGet(Emesh, ownedElemCoords=ownedElemCoords, rc=rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+    allocate(mesh_lons(numOwnedElements))
+    allocate(mesh_lats(numOwnedElements))
+    allocate(mesh_areas(numOwnedElements))
+    do n = 1,npts
+       mesh_lons(n) = ownedElemCoords(2*n-1)
+       mesh_lats(n) = ownedElemCoords(2*n)
+       mesh_areas(n) = 9.85405060010902e-06  ! TODO(mvertens, 2018-11-30) for now hard-wire this for testing
+    end do
+
+    ! obtain CISM internal mesh lats, lons and areas
+    allocate(lats(nx,ny))
+    allocate(lons(nx,ny))
+    allocate(areas(nx,ny))
+
+    allocate(lats_vec(npts))
+    allocate(lons_vec(npts))
+    allocate(areas_vec(npts))
+
+    ! TODO(wjs, 2015-04-02) The following may need a loop over instances
+    call glad_get_lat_lon(ice_sheet, instance_index = 1, lats = lats, lons = lons)
+    call glad_get_areas(ice_sheet, instance_index = 1, areas = areas)
+    call spatial_to_vector(lons, lons_vec)
+    call spatial_to_vector(lats, lats_vec)
+    call spatial_to_vector(areas, areas_vec)
+
+    ! check that areas, lats and lons from the mesh are not different to a tolerance factor
+    ! from areas, lats and lons calculated internally
+
+    areas_vec(:) = areas_vec(:)/(radius*radius) ! convert from m^2 to radians^2
+    do n = 1, npts
+       if ( abs(mesh_lons(n) - lons_vec(n)) > tolerance) then
+          write(6,101),n, lons_vec(n), mesh_lons(n), gindex(n)
+101       format('ERROR: CISM n, lon, mesh_lon, gindex = ',i6,2(f20.10,2x),i8)
+          !          write(6,102) abs(mesh_lons(n) - lons_vec(n))
+          !102       format('ERROR: CISM lon diff = ',f20.10,' is too large')
+          !call shr_sys_abort()
```

Comment:
- [ ] @mvertens - is there a reason that the aborts are commented out for the various checks here  – on lons, lats & areas? Maybe we don't want to compare areas, but should we at least be comparing lats and lons? It seems like either we should uncomment these commented lines or stop comparing lats & lons at all.